### DISCUSSION TITLE
How to calculate GPA and Overall grade of a student of a semeter ?

### DISCUSSION THREAD

**Discussion by Pushpavel** [Pushpavel/project-academics#5] [Q&A] [answered]
while publishing the semester result, we need to calculate GPA and update the semester's summary subcollection

**Comment by LokeshShelva** [comment]
According to the mark , points will be given and the average of the points is GPA I guess. But we should **confirm with sir**

**Reply by Pushpavel** [reply]
so we can't calculate GPA from just total marks of all courses in the semester and overall grade information alone right?

**Comment by moseschand003** [comment, marked as answer]
Let the grade obtained in subject i be G(i) and credit be C(i) then summation of G(i) * C(i) over all courses divided by total credits taken for that semester (two digits precision) is the student's GPA.

Apply this over all courses that we attended during college divided by total credits we took will give our CGPA.

**Reply by Pushpavel** [reply]
how is G(i) a number to multiply?

**Reply by moseschand003** [reply]
S - 10.
A - 9.
and so on...
E - 5.

F - fail.

**Reply by Pushpavel** [reply]
just now updated the question we need to calculate overall grade of a semester too

**Reply by moseschand003** [reply]
There's no overall grade for a semester. I updated the figma file.

**Reply by Pushpavel** [reply]
Ohhh ok, I will update the database model too

**Reply by Pushpavel** [reply]
##### So, we need to update the database model for
1. querying for credits of all courses of a semester enrolled by a batch
1. querying the grades of every student of a batch.

##### Possible solutions
1. by including the credits of all courses a student had enrolled in semester's summary subcollection during initial data import
1. requires us to query every student entries document of a batch (ex-19B1)  so I think we may need a batch field for each student entry

##### Querying for grades of every student of a batch looks like 
```typescript
firestore.collectionGroup('public_student_entries')
.where('sem','==',currentSem)
.where('batch','==',batchId).get()
```

**Reply by moseschand003** [reply]
Nice.
